[[ Bug 22777 ]] Implement hitTest in iOS native container view

montegoulding opened the pull request:
This patch implements a hit test in the iOS native container view so it does not
block events from underlying views. By returning `nil` if the event does not
hit a child view the remaining view hierarchy will be traversed to find the
correct target view.

runrevmark commented on engine/src/native-layer-ios.mm:
```diff
@@ -214,6 +215,19 @@ - (void)setFrame:(CGRect)frame
+{
+	UIView *t_hit_view = [super hitTest:point withEvent:event];
+	
+	/* The container view should not trap any eveents so if the hitTest returns self we
```
`events`

runrevmark commented on engine/src/native-layer-ios.mm:
```diff
@@ -214,6 +215,19 @@ - (void)setFrame:(CGRect)frame
+	
+	/* The container view should not trap any eveents so if the hitTest returns self we
+	 * return nil so the heirarchy continues to be navigated to find a view */
+	if (t_hit_view == self) {
```
Braces go on separate lines.

montegoulding committed:
[[ Bug 22777 ]] Implement hitTest in iOS native container view

This patch implements a hit test in the iOS native container view so it does not
block events from underlying views. By returning `nil` if the event does not
hit a child view the remaining view hierarchy will be traversed to find the
correct target view.

runrevmark commented:
@livecode-vulcan review ok add896c

livecodepanos commented:
@livecode-vulcan review ok add896c

livecodepanos commented:
@livecode-vulcan retry add896c

livecodepanos commented:
@livecode-vulcan retry add896c

livecode-vulcan commented:
💙 review by @livecodepanos ok add896c2a20aecc734a6a86ab2d131c2d040c2c8

livecode-vulcan commented:
😎 test success add896c2a20aecc734a6a86ab2d131c2d040c2c8

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/818)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/227)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/227)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/227)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/227)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2344)
* try-community-universal-ios-iphoneos13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.4/builds/77)
* try-community-universal-ios-iphonesimulator13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.4/builds/79)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2372)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2360)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2359)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2394)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2334)
